var/spack/repos/builtin/packages/openspeedshop/package.py

```diff
@@ -112,48 +110,77 @@ class Openspeedshop(CMakePackage):
-
-    # Dependencies only for the openspeedshop offline package.
-    depends_on("libunwind", when='+offline')
-    depends_on("papi", when='+offline')
-    depends_on("libmonitor+krellpatch", when='+offline')
-    depends_on("openmpi", when='+offline+openmpi')
-    depends_on("mpich", when='+offline+mpich')
-    depends_on("mpich2", when='+offline+mpich2')
-    depends_on("mvapich2", when='+offline+mvapich2')
-    depends_on("mvapich", when='+offline+mvapich')
-    depends_on("mpt", when='+offline+mpt')
-
-    # Dependencies only for the openspeedshop cbtf package.
-    depends_on("cbtf", when='+cbtf')
-    depends_on('cbtf-krell', when='+cbtf')
-    depends_on('cbtf-krell+crayfe', when='+crayfe')
-    depends_on('cbtf-krell+cti', when='+cti')
-    depends_on('cbtf-krell+mpich', when='+cbtf+mpich')
-    depends_on('cbtf-krell+mpich2', when='+cbtf+mpich2')
-    depends_on('cbtf-krell+mpt', when='+cbtf+mpt')
-    depends_on('cbtf-krell+mvapich', when='+cbtf+mvapich')
-    depends_on('cbtf-krell+mvapich2', when='+cbtf+mvapich2')
-    depends_on('cbtf-krell+openmpi', when='+cbtf+openmpi')
-    depends_on("cbtf-argonavis", when='+cbtf+cuda')
-    depends_on("mrnet@5.0.1:+lwthreads", when='+cbtf')
-    depends_on("mrnet@5.0.1:+cti", when='+cti+cbtf')
+
+    # Dependencies for the openspeedshop cbtf packages.
+    depends_on("cbtf@develop", when='@develop')
+    depends_on("cbtf@1.9.1.0", when='@2.3.1.3')
+
+    depends_on("cbtf-krell@develop", when='@develop')
+    depends_on("cbtf-krell@1.9.1.0", when='@2.3.1.3')
+
+    depends_on('cbtf-krell@develop+crayfe', when='@develop+crayfe')
+    depends_on('cbtf-krell@1.9.1.0+crayfe', when='@2.3.1.3+crayfe')
+
+    depends_on('cbtf-krell@develop+cti', when='@develop+cti')
+    depends_on('cbtf-krell@1.9.1.0+cti', when='@2.3.1.3+cti')
+
+    depends_on('cbtf-krell@develop+mpich', when='@develop+mpich')
+    depends_on('cbtf-krell@1.9.1.0+mpich', when='@2.3.1.3+mpich')
+
+    depends_on('cbtf-krell@develop+mpich2', when='@develop+mpich2')
+    depends_on('cbtf-krell@1.9.1.0+mpich2', when='@2.3.1.3+mpich2')
+
+    depends_on('cbtf-krell@develop+mpt', when='@develop+mpt')
+    depends_on('cbtf-krell@1.9.1.0+mpt', when='@2.3.1.3+mpt')
+
+    depends_on('cbtf-krell@develop+mvapich', when='@develop+mvapich')
+    depends_on('cbtf-krell@1.9.1.0+mvapich', when='@2.3.1.3+mvapich')
+
+    depends_on('cbtf-krell@develop+mvapich2', when='@develop+mvapich2')
+    depends_on('cbtf-krell@1.9.1.0+mvapich2', when='@2.3.1.3+mvapich2')
+
+    depends_on('cbtf-krell@develop+openmpi', when='@develop+openmpi')
+    depends_on('cbtf-krell@1.9.1.0+openmpi', when='@2.3.1.3+openmpi')
+
+    depends_on("cbtf-argonavis@develop", when='@develop+cuda')
+    depends_on("cbtf-argonavis@1.9.1.0", when='@2.3.1.3+cuda')
+
+    # For MRNet
+    depends_on("mrnet@5.0.1-3:+cti", when='@develop+cti')
+    depends_on("mrnet@5.0.1-3:+lwthreads", when='@develop')
+
+    depends_on("mrnet@5.0.1-3:+cti", when='@2.3.1.3+cti')
```

Comment:
@jgalarowicz did you mean `mrnet@5.0.1-3+cti` vs. `mrnet@5.0.1-3:+cti`? i.e. did you want to be exact about the version? If not, these two lines can be merged with the above two lines.

Reply:
@scheibelp Yes, we wanted the exact version for the release type path through the O|SS packages.    The release would be more stable based on using packages we know to work with the rest of the tool.